diff -u a/vendor/oplus/kernel/mm/hybridswap_zram/Kconfig b/vendor/oplus/kernel/mm/hybridswap_zram/Kconfig
--- a/vendor/oplus/kernel/mm/hybridswap_zram/Kconfig
+++ b/vendor/oplus/kernel/mm/hybridswap_zram/Kconfig
@@ -3,6 +3,9 @@ config HYBRIDSWAP_ZRAM
 	tristate "Compressed RAM block device support"
 	depends on BLOCK && SYSFS && (THP_ZSMALLOC||ZSMALLOC) && CRYPTO && !ZRAM
 	select CRYPTO_LZO
+	select CRYPTO_LZ4K
+	select CRYPTO_LZ4KD
+	select CRYPTO_842
 	default n
 	help
 	  Creates virtual block devices called /dev/zramX (X = 0, 1, ...).

diff -u a/vendor/oplus/kernel/mm/hybridswap_zram/zcomp.c b/vendor/oplus/kernel/mm/hybridswap_zram/zcomp.c
--- a/vendor/oplus/kernel/mm/hybridswap_zram/zcomp.c
+++ b/vendor/oplus/kernel/mm/hybridswap_zram/zcomp.c
@@ -26,6 +26,12 @@ static const char * const backends[] = {
 #if IS_ENABLED(CONFIG_CRYPTO_LZ4HC)
 	"lz4hc",
 #endif
+#if IS_ENABLED(CONFIG_CRYPTO_LZ4K)
+	"lz4k",
+#endif
+#if IS_ENABLED(CONFIG_CRYPTO_LZ4KD)
+	"lz4kd",
+#endif
 #if IS_ENABLED(CONFIG_CRYPTO_842)
 	"842",
 #endif
